From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Tue, 7 Mar 2023 11:09:59 +0000
Subject: [PATCH] Allow disabling the partner name

Add options to set or disable partner name.

diff --git a/google-beta/provider/provider.go b/google-beta/provider/provider.go
index 36347670d..6fe79c98d 100644
--- a/google-beta/provider/provider.go
+++ b/google-beta/provider/provider.go
@@ -170,6 +170,19 @@ func Provider() *schema.Provider {
 				Optional: true,
 			},
 
+			"google_partner_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"disable_google_partner_name": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+					"GOOGLE_DISABLE_PARTNER_NAME",
+				}, nil),
+			},
+
 			"request_reason": {
 				Type:     schema.TypeString,
 				Optional: true,
@@ -1110,6 +1123,21 @@ func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Pr
 		UserAgent:           p.UserAgent("terraform-provider-google-beta", version.ProviderVersion),
 	}
 
+	disablePartnerName := d.Get("disable_google_partner_name").(bool)
+	userSpecifiedPartnerName := d.Get("google_partner_name")
+
+	partnerString := ""
+	if !disablePartnerName {
+		if userSpecifiedPartnerName != "" {
+			partnerString = fmt.Sprintf("GPN:%s; ", userSpecifiedPartnerName)
+		} else {
+			partnerString = "GPN:Pulumi; "
+		}
+	}
+	userAgent := fmt.Sprintf("Pulumi/3.0 (%shttps://www.pulumi.com) pulumi-gcp/%s", partnerString, version.ProviderVersion)
+
+	config.UserAgent = userAgent
+
 	// opt in extension for adding to the User-Agent header
 	if ext := os.Getenv("GOOGLE_TERRAFORM_USERAGENT_EXTENSION"); ext != "" {
 		ua := config.UserAgent
