From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Thu, 22 Jun 2023 16:38:12 -0700
Subject: [PATCH] Remove duplicative resource token


diff --git a/google-beta/provider/provider.go b/google-beta/provider/provider.go
index 225df27918..d899e3d14b 100644
--- a/google-beta/provider/provider.go
+++ b/google-beta/provider/provider.go
@@ -249,7 +249,7 @@ func Provider() *schema.Provider {
 			},
 		},
 
-		DataSourcesMap: registry.DatasourceMap(),
+		DataSourcesMap: DatasourceMap(),
 		ResourcesMap:   registry.ResourceMap(),
 	}
 
@@ -267,14 +267,21 @@ func Provider() *schema.Provider {
 	return provider
 }
 
-// Compatibility shim for diff-processor.
 func ResourceMap() map[string]*schema.Resource {
 	return registry.ResourceMap()
 }
 
 // Compatibility shim for diff-processor.
 func DatasourceMap() map[string]*schema.Resource {
-	return registry.DatasourceMap()
+	datasourceMap := registry.DatasourceMap()
+	datasourcesToRemove := []string{
+		"google_organization_iam_policy",
+		"google_organization_iam_policy_legacy",
+	}
+	for _, datasourceToRemove := range datasourcesToRemove {
+		delete(datasourceMap, datasourceToRemove)
+	}
+	return datasourceMap
 }
 
 func ProviderConfigure(ctx context.Context, d *schema.ResourceData, p *schema.Provider) (interface{}, diag.Diagnostics) {
