From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Thu, 7 Sep 2023 16:28:00 -0700
Subject: [PATCH] Fix #794 with an unconditional read.


diff --git a/google-beta/services/sql/resource_sql_database_instance.go b/google-beta/services/sql/resource_sql_database_instance.go
index 7de93c2a5..055ebc7c7 100644
--- a/google-beta/services/sql/resource_sql_database_instance.go
+++ b/google-beta/services/sql/resource_sql_database_instance.go
@@ -2158,10 +2158,11 @@ func resourceSqlDatabaseInstanceUpdate(d *schema.ResourceData, meta interface{})
 		if err != nil {
 			return err
 		}
-		err = resourceSqlDatabaseInstanceRead(d, meta)
-		if err != nil {
-			return err
-		}
+	}
+
+	err = resourceSqlDatabaseInstanceRead(d, meta)
+	if err != nil {
+		return err
 	}
 
 	s := d.Get("settings")
