From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Thu, 22 Jun 2023 16:38:12 -0700
Subject: [PATCH 08/10] Remove duplicative resource token


diff --git a/google-beta/provider/provider.go b/google-beta/provider/provider.go
index 7b195cd93..dc2771db9 100644
--- a/google-beta/provider/provider.go
+++ b/google-beta/provider/provider.go
@@ -793,6 +793,15 @@ func Provider() *schema.Provider {
 
 func DatasourceMap() map[string]*schema.Resource {
 	datasourceMap, _ := DatasourceMapWithErrors()
+
+	datasourcesToRemove := []string{
+		"google_organization_iam_policy",
+		"google_organization_iam_policy_legacy",
+	}
+	for _, datasourceToRemove := range datasourcesToRemove {
+		delete(datasourceMap, datasourceToRemove)
+	}
+
 	return datasourceMap
 }
 
