From 1c586ae62a4a7533945d70f5374d4bea18abeaba Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Thu, 22 Jun 2023 16:38:12 -0700
Subject: [PATCH 9/9] Remove duplicative resource token

---
 google-beta/provider.go | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/google-beta/provider.go b/google-beta/provider.go
index 52246cbe5..063721413 100644
--- a/google-beta/provider.go
+++ b/google-beta/provider.go
@@ -804,6 +804,15 @@ func Provider() *schema.Provider {
 
 func DatasourceMap() map[string]*schema.Resource {
 	datasourceMap, _ := DatasourceMapWithErrors()
+
+	datasourcesToRemove := []string{
+		"google_organization_iam_policy",
+		"google_organization_iam_policy_legacy",
+	}
+	for _, datasourceToRemove := range datasourcesToRemove {
+		delete(datasourceMap, datasourceToRemove)
+	}
+
 	return datasourceMap
 }
 
-- 
2.41.0

