From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Fri, 30 Aug 2024 14:01:35 -0700
Subject: [PATCH] Rename default provisioning label to goog-pulumi-provisioned


diff --git a/google-beta/transport/config.go b/google-beta/transport/config.go
index 54c9df86d..b7f7fa52a 100644
--- a/google-beta/transport/config.go
+++ b/google-beta/transport/config.go
@@ -797,7 +797,7 @@ var DefaultClientScopes = []string{
 	"https://www.googleapis.com/auth/userinfo.email",
 }
 
-const AttributionKey = "goog-terraform-provisioned"
+const AttributionKey = "goog-pulumi-provisioned"
 const AttributionValue = "true"
 const CreateOnlyAttributionStrategy = "CREATION_ONLY"
 const ProactiveAttributionStrategy = "PROACTIVE"
