From ad3f2a2e6456d5fad26a774fac5797c661a68c9b Mon Sep 17 00:00:00 2001
From: Aaron Friel <mayreply@aaronfriel.com>
Date: Thu, 22 Jun 2023 16:38:12 -0700
Subject: [PATCH 10/11] Remove duplicative resource token

---
 google-beta/provider/provider.go | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/google-beta/provider/provider.go b/google-beta/provider/provider.go
index 9d0614536..05afbe87f 100644
--- a/google-beta/provider/provider.go
+++ b/google-beta/provider/provider.go
@@ -868,6 +868,15 @@ func Provider() *schema.Provider {
 
 func DatasourceMap() map[string]*schema.Resource {
 	datasourceMap, _ := DatasourceMapWithErrors()
+
+	datasourcesToRemove := []string{
+		"google_organization_iam_policy",
+		"google_organization_iam_policy_legacy",
+	}
+	for _, datasourceToRemove := range datasourcesToRemove {
+		delete(datasourceMap, datasourceToRemove)
+	}
+
 	return datasourceMap
 }
 
-- 
2.40.0

