From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: guineveresaenger <guinevere@pulumi.com>
Date: Fri, 6 Sep 2024 16:24:54 -0700
Subject: [PATCH] make sure we read the label?


diff --git a/google-beta/tpgresource/labels.go b/google-beta/tpgresource/labels.go
index 9f7a25d88..d0f5c3e97 100644
--- a/google-beta/tpgresource/labels.go
+++ b/google-beta/tpgresource/labels.go
@@ -45,33 +45,39 @@ func SetLabelsNoDefaults(labels map[string]string, d *schema.ResourceData, linea
 	q.Q(labels)
 	q.Q(defaultLabels)
 	q.Q(lineage)
-	if v, ok := d.GetOk(lineage); ok {
+	if v, ok := d.GetOk(lineage); ok { // TODO: this is a check to see if the resource data has these keys - "labels" or "terraform_labels".
 		q.Q("just reading")
 		// We are reading after an update, so populate just the user defined labels.
 		if labels != nil {
 			for k := range v.(map[string]interface{}) {
-				// This label is set by the provider.
-				// It is neither a defaultLabel, nor is it user set. Skip.
-				if k == "goog-pulumi-provisioned" && lineage == "terraform_labels" {
-					q.Q("reading after update, found goog label")
-					continue
-				}
+				//// This label is set by the provider.
+				//// It is neither a defaultLabel, nor is it user set. Skip.
+				//if k == "goog-pulumi-provisioned" && lineage == "terraform_labels" {
+				//	q.Q("reading after update, found goog label")
+				//	continue
+				//}
 				transformed[k] = labels[k]
 			}
 		}
 	} else {
 		q.Q("importing")
-		// We are reading for an import, so populate all of the labels, except the skipped ones.
+		// We are reading for an import, so populate all of the labels, except the skipped ones. // TODO: what does "import" here mean?
 		for k, v := range labels {
-			if _, ok := defaultLabels[k]; !ok || lineage != "labels" {
+			if _, ok := defaultLabels[k]; !ok || lineage != "labels" { // TODO: this is happening all the time, so there's something busted here.
 				q.Q("we did NOT find k in default labels OR lineage was NOT labels", k, lineage)
-				if k == "goog-pulumi-provisioned" && lineage == "terraform_labels" {
-					q.Q("we should actually check for lineage here too I think. But this doesn't explain why default labels made it in???")
-					continue
-				}
+				//if k == "goog-pulumi-provisioned" && lineage == "terraform_labels" {
+				//	q.Q("we should actually check for lineage here too I think. But this doesn't explain why default labels made it in???")
+				//	continue
+				//}
+				transformed[k] = v
+			}
+			// TODO: the goog-pulumi-provisioned label is NEVER in defaultLabels! So we're just dropping it on accident.
+			if k == "goog-pulumi-provisioned" {
 				transformed[k] = v
 			}
 		}
+		//tODO: I think when we're "importing", whatever this means - this is actually not what is going on unless we're overloading the word import here somehow -
+		// we NEVER read the goog-pulumi-provisioned label. And we kinda should. in the basic case, importing is always on labels, never on terraform_labels
 	}
 	q.Q(transformed)
 
